libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/generator/OpenHouseDataLayoutStrategyGenerator.java

```diff
@@ -62,12 +62,21 @@ private DataLayoutStrategy generateCompactionStrategy() {
     Dataset<Long> fileSizes =
         tableFileStats.get().map((MapFunction<FileStat, Long>) FileStat::getSize, Encoders.LONG());
 
+    Dataset<Long> filteredSizes =
+        fileSizes.filter(
+            (FilterFunction<Long>)
+                size ->
+                    size < TARGET_BYTES_SIZE * DataCompactionConfig.MIN_BYTE_SIZE_RATIO_DEFAULT);
+    // Check whether we have anything to map/reduce on for cost computation, this is only the case
+    // if we have small files that need to be compacted.
+    // Return default (empty) object.
+    if (filteredSizes.count() == 0) {
+      return DataLayoutStrategy.builder().config(null).cost(0).gain(0).score(0).entropy(0).build();
```

Comment:
It might be cleaner to return Optional<DataLayoutStrategy>, easier to handle outside, we should probably just skip it

Reply:
Done!